### PR TITLE
ACCUMULO-4528 Accumulo scripts improvements

### PR DESCRIPTION
**Pull request by mikewalch**
* Accumulo now determines hostname in Java
* Minimized use of environment variables
* Consolidated scripts by moving service.sh and cluster.sh into
  accumulo-service and accumulo-cluster which were moved to contrib/.
  Also, moved masters, tservers, tracers, etc files to contrib/conf/
* Removed accumulo-watcher script as restarting services should not be
  handled by Accumulo scripts
* NUMA and multiple tservers are no longer configured in scripts but
  could be NUMA commands could be added using ACCUMULO_JAVA_PREFIX env
  variable.
* Renamed accumulo-env.sh to accumulo.conf
* Moved create-config and build-native from accumulo script to
  accumulo-config and accumulo-native scripts in contrib/

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -57,15 +57,15 @@ The script will ask you questions about your set up. Below are some suggestions:
...
+After the `accumulo-config` command is run, the `conf/` directory will contain
+`accumulo.conf`, `accumulo-site.xml`, and few a additional files. These files require
 a few edits before starting Accumulo.
 
 ### Secret
 
 Accumulo coordination and worker processes can only communicate with each other
 if they share the same secret key.  To change the secret key set
-`instance.secret` in `conf/accumulo-site.xml`.  Changing this secret key from
+`instance.secret` in `accumulo-site.xml`.  Changing this secret key from
```
Comment (milleruntime) [inline INSTALL.md]: Did you want to move accumulo-site.xml out of conf?

Comment (mikewalch) [inline INSTALL.md]: It's still located in conf for the tarball distribution but I changed the manual to not include directory/path of the file as accumulo-site.xml could be in other locations like `/etc/accumulo` in downstream packaging.

File: assemble/contrib/tool.sh [hunk 2]
```diff
@@ -26,14 +26,21 @@ contrib=$( cd -P "$( dirname "$SOURCE" )" && pwd )
 basedir=$( cd -P "${contrib}"/.. && pwd )
 # Stop: Resolve Script Directory
 
-source "$basedir"/libexec/load-env.sh
+conf="${basedir}/conf"
+if [ ! -f "${conf}/accumulo.conf" ]; then
+  echo "accumulo.conf must exist in $conf"
```
Comment (ctubbsii) [inline assemble/contrib/tool.sh]: Why wouldn't this file be optional?

File: assemble/contrib/tool.sh [hunk 3]
```diff
@@ -26,14 +26,21 @@ contrib=$( cd -P "$( dirname "$SOURCE" )" && pwd )
...
+  echo "Run 'accumulo-config' to create it or copy it from $conf/examples"
+  echo "Follow the instructions in INSTALL.md to edit it for your environment."
+  exit 1
+fi
+source "$conf/accumulo.conf"  
 
 if [[ -z "$HADOOP_PREFIX" ]] ; then
-   echo "HADOOP_PREFIX is not set.  Please make sure it's set globally or in conf/accumulo-env.sh"
+   echo "HADOOP_PREFIX is not set.  Please make sure it's set globally or in $conf/accumulo.conf"
```
Comment (ctubbsii) [inline assemble/contrib/tool.sh]: I think this is currently necessary for the property interpolation in the `accumulo-site.xml` file, but I'd really like to phase out having it as a requirement.

File: INSTALL.md [hunk 4]
```diff
@@ -25,11 +25,11 @@ source code.  Unpack as follows.
...
     cd accumulo-X.Y.Z
 
-There are three scripts in the the `bin/` directory that are used to manage Accumulo:
+There are three scripts in the tarball distribution that are used to manage Accumulo:
 
-1. `accumulo` - Runs Accumulo command-line tools and starts Accumulo processes
-2. `accumulo-service` - Runs Accumulo processes as services
-3. `accumulo-cluster` - Manages Accumulo cluster on a single node or several nodes
+1. `bin/accumulo` - Runs Accumulo command-line tools and starts Accumulo processes
+2. `contrib/accumulo-service` - Runs Accumulo processes as services
```
Comment (keith-turner) [inline INSTALL.md]: What does it mean for a script to be in contrib vs bin?

Comment (ctubbsii) [inline INSTALL.md]: In my view, `contrib/` should be for add-ons/wrappers which could be separate projects on their own, but which we are currently maintaining to provide convenient functionality.

I think it makes sense to have a basic launch script in `bin/`, which minimally wraps the execution of java. That minimal wrapper wouldn't make sense as its own project and thus (by my opinion) shouldn't be in `contrib/`.

I'm far less opinionated on the convenience scripts for producing the example configs and for executing `make` on the unpacked native tarball. They could probably go either place.

Comment (mikewalch) [inline INSTALL.md]: I thought about having a contrib and a bin directory. I think there should only be a bin/ directory.  Two directories will confuse people.  We can communicate that scripts are optional/add-on in the documentation rather than the directory structure.  To limit the number of scripts in bin, we can combine many of the scripts in contrib into one or two scripts.  If a script doesn't work or we don't want to support it any more the in the tarball distribution, it should either be removed or move to a contrib directory in the repo that doesn't get added to the distribution. Contrib scripts should not be distributed.  As for the scripts in bin/, downstream packagers can pick and choose what they include.

Comment (ctubbsii) [inline INSTALL.md]: Seems reasonable.

File: INSTALL.md (hunk 1 above)
Comment (milleruntime) [inline INSTALL.md]: Got it.  I took the changes out of context.

**Comment by milleruntime** [comment]
+1 On the renaming of accumulo-env.sh to accumulo.conf.  The .conf extension is pretty clear what the file is for, its simply for configuration purposes.  The -env.sh extension tells me this shell script is setting up the environment but could be doing more then configuration, which makes me nervous just seeing the name of the file.  From a user perspective, I would be much more inclined and comfortable to modify a .conf file.

File: INSTALL.md [hunk 5]
```diff
@@ -40,13 +40,13 @@ Accumulo has some optional native code that improves its performance and
 stability. Before configuring Accumulo, attempt to build this native code
 with the following command.
 
-    accumulo build-native
+    ./contrib/accumulo-native
```
Comment (mikewalch) [inline INSTALL.md]: I pushed a commit where I moved the command to `bin/accumulo-util build-native`

File: assemble/contrib/tool.sh (hunk 2 above)
Comment (mikewalch) [inline assemble/contrib/tool.sh]: I made it optional in 7f9da9f12ce5940

File: INSTALL.md [hunk 6]
```diff
@@ -40,13 +40,13 @@ Accumulo has some optional native code that improves its performance and
...
 
 If the command fails, its OK to continue with setup and resolve the issue later.
 
 Run the command below to create configuration for Accumulo in `conf/`:
 
-    accumulo create-config
+    ./contrib/accumulo-config
```
Comment (mikewalch) [inline INSTALL.md]: This was moved to `bin/accumulo-util create-config`

File: assemble/conf/examples/generic_logger.properties [hunk 7]
```diff
@@ -15,7 +15,7 @@
 
 # Write out everything at the DEBUG level to the debug log
 log4j.appender.A2=org.apache.log4j.RollingFileAppender
-log4j.appender.A2.File=${org.apache.accumulo.core.dir.log}/${org.apache.accumulo.core.application}_${accumulo.service.instance}${org.apache.accumulo.core.ip.localhost.hostname}_fromprops.debug.log
+log4j.appender.A2.File=${org.apache.accumulo.core.dir.log}/${org.apache.accumulo.core.application}${accumulo.service.instance}_${org.apache.accumulo.core.ip.localhost.hostname}_fromprops.debug.log
```
Comment (mikewalch) [inline assemble/conf/examples/generic_logger.properties]: This should be done in a separate PR.  Created issue https://issues.apache.org/jira/browse/ACCUMULO-4588

File: assemble/conf/examples/generic_logger.properties [hunk 8]
```diff
@@ -24,7 +24,7 @@ log4j.appender.A2.layout.ConversionPattern=%d{ISO8601} [%-8c{2}] %-5p: %m%n
 
 # Write out INFO and higher to the regular log
 log4j.appender.A3=org.apache.log4j.RollingFileAppender
-log4j.appender.A3.File=${org.apache.accumulo.core.dir.log}/${org.apache.accumulo.core.application}_${accumulo.service.instance}${org.apache.accumulo.core.ip.localhost.hostname}_fromprops.log
+log4j.appender.A3.File=${org.apache.accumulo.core.dir.log}/${org.apache.accumulo.core.application}${accumulo.service.instance}_${org.apache.accumulo.core.ip.localhost.hostname}_fromprops.log
```
Comment (mikewalch) [inline assemble/conf/examples/generic_logger.properties]: This should be done in a separate PR.  Created issue https://issues.apache.org/jira/browse/ACCUMULO-4588

File: INSTALL.md [hunk 9]
```diff
@@ -76,19 +76,18 @@ When configuring Accumulo the following information about these dependencies
...
+   in `accumulo-site.xml`.
  * **Where to store data** :  Provide this by setting `instance.volumes` in
-   `conf/accumulo-site.xml`.  If your namenode is running at 192.168.1.9:9000
+   `accumulo-site.xml`.  If your namenode is running at 192.168.1.9:9000
    and you want to store data in `/accumulo` in HDFS, then set
   `instance.volumes` to `hdfs://192.168.1.9:9000/accumulo`.
  * **Location of Zoookeeper and Hadoop jars** :  Setting `ZOOKEEPER_HOME` and
-   `HADOOP_PREFIX` in `conf/accumulo-env.sh` will help Accumulo find these
-   jars.
+   `HADOOP_PREFIX` in `accumulo.conf` will help Accumulo find these jars.
```
Comment (mikewalch) [inline INSTALL.md]: Improve documentation in 1e8d87cd311

**Comment by mikewalch** [comment]
Thanks everyone for reviewing!  Your feedback resulted in a lot of changes that made this PR better.  I think I have addressed all comments.  Unless someone needs more time to review or wants more changes, I plan on squashing and merging by end of today.

**Comment by mikewalch** [comment]
I talked to @milleruntime offline about accumulo-env.sh rename.  For this PR, I decided to not do the rename as concerns were brought up about changing name to accumulo.conf.  In the end, I would like a better name for accumulo-env.sh but I don't want to force a change until we find a better name.
